common/lib/xmodule/xmodule/vertical_block.py

```diff
@@ -52,6 +52,8 @@ def student_view(self, context):
 
         child_context['child_of_vertical'] = True
 
+        is_child_of_vertical = context.get('child_of_vertical', False)
```

Comment:
I'm not certain this would be propagated down all descendants - especially if non-verticals are within the hierarchy.